### DISCUSSION TITLE
No connection on Unraid

### DISCUSSION THREAD

**Discussion by swaggerino** [ghostfolio/ghostfolio#1762] [Q&A]
**Bug Description**
I'm suddenly getting all those messages in the log and my ghostfolio is not longer showing any graphs etc.
When I click the link in red, the connection works fine so something must be wrong with accessing it by ghostfolio
(link https://query2.finance.yahoo.com/v7/finance/quote?symbols=22UA.F%2C27W.F%2C2B76.DE%2C2PP.DE%2C3CP.F%2C3V64.DE%2C6WS.F%2C8CF.BE%2CABEC.DE%2CAFX.DE%2CAIY.DE%2CALV.DE%2CAMD.DE%2CAMEM.F%2CCBGB.F%2CCCC3.DE%2CCIS.DE%2CDBAN.DE%2CEUNL.DE%2CEXS3.DE%2CHEIA.AS%2CHNR1.DE%2CHOT.DE%2CILU.DE%2CINL.DE%2CIQQ6.DE%2CIQQP.F%2CISPA.DE%2CLAR.DE%2CLIN.DE%2CLYMS.DE%2CM4I.DE%2CMDO.DE%2CMMM.DE%2CMOR.DE%2CMSF.DE%2CMUV2.DE%2CMXI.DE%2CNVD.DE%2CPEP.DE%2CQDVG.DE%2CRY6.DE%2CSAP.DE%2CSXR8.DE%2CWMT.DE )

**Expected behavior**
It used to work fine but I didn't keep track of the version when it worked last time.

**Logs**

```
[Nest] 280  - 02/28/2023, 5:37:43 PM     LOG    ________               __  ____      ___
[Nest] 280  - 02/28/2023, 5:37:43 PM     LOG   / ____/ /_  ____  _____/ /_/ __/___  / (_)___
[Nest] 280  - 02/28/2023, 5:37:43 PM     LOG  / / __/ __ \/ __ \/ ___/ __/ /_/ __ \/ / / __ \
[Nest] 280  - 02/28/2023, 5:37:43 PM     LOG / /_/ / / / / /_/ (__  ) /_/ __/ /_/ / / / /_/ /
[Nest] 280  - 02/28/2023, 5:37:43 PM     LOG \____/_/ /_/\____/____/\__/_/  \____/_/_/\____/ v1.240.0
[Nest] 280  - 02/28/2023, 5:37:43 PM     LOG 
[Nest] 280  - 02/28/2023, 5:37:43 PM     LOG Listening at http://0.0.0.0:3333
[Nest] 280  - 02/28/2023, 5:37:43 PM     LOG 
[Nest] 280  - 02/28/2023, 5:39:55 PM   ERROR [YahooFinanceService] FetchError: request to https://query2.finance.yahoo.com/v7/finance/quote?symbols=EURUSD%3DX failed, reason: connect ETIMEDOUT 2a00:1288:80:807::1:443
[Nest] 280  - 02/28/2023, 5:39:55 PM   ERROR [YahooFinanceService] FetchError: request to https://query2.finance.yahoo.com/v7/finance/quote?symbols=EURUSD%3DX failed, reason: connect ETIMEDOUT 2a00:1288:80:807::1:443
[Nest] 280  - 02/28/2023, 5:40:11 PM   ERROR [YahooFinanceService] FetchError: request to https://query2.finance.yahoo.com/v7/finance/quote?symbols=22UA.F%2C27W.F%2C2B76.DE%2C2PP.DE%2C3CP.F%2C3V64.DE%2C6WS.F%2C8CF.BE%2CABEC.DE%2CAFX.DE%2CAIY.DE%2CALV.DE%2CAMD.DE%2CAMEM.F%2CCBGB.F%2CCCC3.DE%2CCIS.DE%2CDBAN.DE%2CEUNL.DE%2CEXS3.DE%2CHEIA.AS%2CHNR1.DE%2CHOT.DE%2CILU.DE%2CINL.DE%2CIQQ6.DE%2CIQQP.F%2CISPA.DE%2CLAR.DE%2CLIN.DE%2CLYMS.DE%2CM4I.DE%2CMDO.DE%2CMMM.DE%2CMOR.DE%2CMSF.DE%2CMUV2.DE%2CMXI.DE%2CNVD.DE%2CPEP.DE%2CQDVG.DE%2CRY6.DE%2CSAP.DE%2CSXR8.DE%2CWMT.DE failed, reason: connect ETIMEDOUT 2a00:1288:80:807::2:443
[Nest] 280  - 02/28/2023, 5:40:11 PM    WARN [PortfolioCalculator] Missing initial value for symbol 27W.F at 2020-10-28
[Nest] 280  - 02/28/2023, 5:40:11 PM    WARN [PortfolioCalculator] Missing initial value for symbol 2B76.DE at 2020-12-02
[Nest] 280  - 02/28/2023, 5:40:11 PM    WARN [PortfolioCalculator] Missing initial value for symbol 3CP.F at 2020-12-02
[...]
[Nest] 280  - 02/28/2023, 5:40:11 PM    WARN [PortfolioCalculator] Missing initial value for symbol SXR8.DE at 2021-04-06
[Nest] 280  - 02/28/2023, 5:40:11 PM    WARN [PortfolioCalculator] Missing initial value for symbol WMT.DE at 2020-10-16
[Nest] 280  - 02/28/2023, 5:40:20 PM   ERROR [YahooFinanceService] FetchError: request to https://query2.finance.yahoo.com/v7/finance/quote?symbols=22UA.F%2C27W.F%2C2B76.DE%2C2PP.DE%2C3CP.F%2C3V64.DE%2C6WS.F%2C8CF.BE%2CABEC.DE%2CAFX.DE%2CAIY.DE%2CALV.DE%2CAMD.DE%2CAMEM.F%2CCBGB.F%2CCCC3.DE%2CCIS.DE%2CDBAN.DE%2CEUNL.DE%2CEXS3.DE%2CHEIA.AS%2CHNR1.DE%2CHOT.DE%2CILU.DE%2CINL.DE%2CIQQ6.DE%2CIQQP.F%2CISPA.DE%2CLAR.DE%2CLIN.DE%2CLYMS.DE%2CM4I.DE%2CMDO.DE%2CMMM.DE%2CMOR.DE%2CMSF.DE%2CMUV2.DE%2CMXI.DE%2CNVD.DE%2CPEP.DE%2CQDVG.DE%2CRY6.DE%2CSAP.DE%2CSXR8.DE%2CWMT.DE failed, reason: connect ETIMEDOUT 2a00:1288:80:807::2:443
[Nest] 280  - 02/28/2023, 5:40:20 PM    WARN [PortfolioCalculator] Missing initial value for symbol 27W.F at 2020-10-28
[Nest] 280  - 02/28/2023, 5:40:20 PM    WARN [PortfolioCalculator] Missing initial value for symbol 2B76.DE at 2020-12-02
[Nest] 280  - 02/28/2023, 5:40:20 PM    WARN [PortfolioCalculator] Missing initial value for symbol 3CP.F at 2020-12-02
[Nest] 280  - 02/28/2023, 5:40:20 PM    WARN [PortfolioCalculator] Missing initial value for symbol 3V64.DE at 2019-04-11
[Nest] 280  - 02/28/2023, 5:40:20 PM    WARN [PortfolioCalculator] Missing initial value for symbol 6WS.F at 2021-09-17
[...]
[Nest] 280  - 02/28/2023, 5:40:20 PM    WARN [PortfolioCalculator] Missing initial value for symbol NVD.DE at 2020-02-24
[Nest] 280  - 02/28/2023, 5:40:20 PM    WARN [PortfolioCalculator] Missing initial value for symbol PEP.DE at 2020-12-16
[Nest] 280  - 02/28/2023, 5:40:20 PM    WARN [PortfolioCalculator] Missing initial value for symbol QDVG.DE at 2020-12-02
[Nest] 280  - 02/28/2023, 5:40:20 PM    WARN [PortfolioCalculator] Missing initial value for symbol RY6.DE at 2020-10-14
[Nest] 280  - 02/28/2023, 5:40:20 PM    WARN [PortfolioCalculator] Missing initial value for symbol SXR8.DE at 2021-04-06
[Nest] 280  - 02/28/2023, 5:40:20 PM    WARN [PortfolioCalculator] Missing initial value for symbol WMT.DE at 2020-10-16
```

**Environment**

- Ghostfolio Version 1.240.0
- Browser: Firefox/Chrome/Edge
- OS: unraid

**Comment by dtslvr** [comment]
Hi @swaggerino

Which version of Ghostfolio are you using? Does `FetchError` occur permanently or only temporarily?

Could you please follow the template for future bug reports to make it easier to investigate and fix issues? Especially the version and log as plain text would be useful. Thank you.

**Comment by swaggerino** [comment]
sorry, I've editied the inital post accordingly
Ghostfolio version is 1.240.0.
That error occurs permanently and when I go to the _Admin control_ the log spits out another error:
`[Nest] 280  - 02/28/2023, 5:49:46 PM   ERROR [ExchangeRateDataService] No exchange rate has been found for EURUSD`

Edit:
I remember I had to change my redis to be password protected for another container. This could have been the moment when ghostfolio did no longer work, not sure though.
I did however add/set the REDIS_PASSWORD variable in ghostfolio.

**Comment by theiamdude** [comment]
I am seeing a similar behaviour on my new setup with slightly different error:
`ERROR [YahooFinanceService] FetchError: request to https://query2.finance.yahoo.com/v7/finance/quote?symbols=USDAUD%3DX%2CUSDEUR%3DX%2CUSDGBP%3DX failed, reason: getaddrinfo EAI_AGAIN query2.finance.yahoo.com`

I have just started using Ghostfolio using docker for the first time, it was working ok in the initial run but when I updated the platform table with some new entries I started observing the error and I am not able to get accurate report of the portfolio .

Below are some of the errors I am seeing:
```
[Nest] 250  - 03/01/2023, 9:19:39 AM   ERROR [ExchangeRateDataService] No exchange rate has been found for USDAUD
[Nest] 250  - 03/01/2023, 9:19:39 AM   ERROR [ExchangeRateDataService] No exchange rate has been found for AUDUSD
```
`
[Nest] 250  - 03/01/2023, 9:19:39 AM   ERROR [YahooFinanceService] FetchError: request to https://query2.finance.yahoo.com/v7/finance/quote?symbols=AT1.AX%2CHTA.AX%2CKGN.AX%2CNAB.AX%2CTLS.AX%2CTRS.AX%2CVHY.AX failed, reason: getaddrinfo EAI_AGAIN query2.finance.yahoo.com`

```
[Nest] 250  - 03/01/2023, 9:19:28 AM    WARN [PortfolioCalculator] Missing initial value for symbol HTA.AX at 2018-08-29
[Nest] 250  - 03/01/2023, 9:19:29 AM    WARN [PortfolioCalculator] Missing initial value for symbol NAB.AX at 2020-01-15
[Nest] 250  - 03/01/2023, 9:19:29 AM    WARN [PortfolioCalculator] Missing initial value for symbol TLS.AX at 2018-06-17
[Nest] 250  - 03/01/2023, 9:19:29 AM    WARN [PortfolioCalculator] Missing initial value for symbol VHY.AX at 2020-03-03
```

--- EDIT ---
After further troubleshooting, I found out that having AdGuard setup as a DNS in my network was causing the errors above. Still not sure while it would be blocked and why it would have worked initially, but once I configure 8.8.8.8 as DNS for the docker image, it started working without issues so far.

**Comment by swaggerino** [comment]
I know adguard as DNS can mess with container's ability to reach the internet so I almost always configure my dockers with `--dns 1.0.0.1 --dns 1.1.1.1` 
Ghostfolio is configured like that aswell but I also just tried 8.8.8.8 and turning off the protection in adguard, both things didn't solve the error unfortunately

Could it be my authelia that I've got infront of Ghostfolio?

**Comment by swaggerino** [comment]
disabled authelia, error persists

**Comment by dtslvr** [comment]
Have you noticed the same problems @LeonStoldt?

**Comment by swaggerino** [comment]
Hey guys, just updated the container to 1.242.0 and everything is working fine again!
